config/boards/nanopi-r4s.wip

```diff
@@ -0,0 +1,10 @@
+# Rockchip RK3399 hexa core 4GB RAM SoC 2 x GBE USB3 USB-C
+BOARD_NAME="NanoPi M4V2"
```

Comment:
This should read `BOARD_NAME="NanoPi R4S"` instead. And FriendlyELEC also sells a 1 GB variant.

Reply:
Somewhere I read the 1GB variant is DDR3 while the 4GB is LPDDR4. Might be a dumb question but does this make a difference in the DT? So it may need to be split up into two images?

Reply:
It does say so in the [wiki](https://wiki.friendlyarm.com/wiki/index.php/NanoPi_R4S#Hardware_Spec):
_RAM: 1GB DDR3/4GB LPDDR4_

Reply:
> Might be a dumb question but does this make a difference in the DT? So it may need to be split up into two images?

Rockchip ddr loader supports both types of memory in a single image and that's what we are using for most of the rk3399 family.

Unless we support the board in legacy and enable dmc for it we should be able to get on with a single DT and image for both models. FriendlyElec also [uses a single DT](https://github.com/friendlyarm/kernel-rockchip/commit/dd2a653878f59f596fb68b1e7938b3ab020b72d2) for 5.4.y

I am leaning towards not supporting it in legacy ;-) given that it has no multimedia capabalities.
If we decide to do it anyway we can provide overlay to tweak frequencies per model.

Reply:
> I am leaning towards not supporting it in legacy 

Agree. Not sure if there is anything left for CLI ... porting Rockchip random generator perhaps? If that makes some value.

Reply:
It COULD still be used as streaming server and transcoding via videoengine would bring great performance increase.

Reply:
> porting Rockchip random generator perhaps?

I am planning to do it at some point. Just too many life chores lately...

> It COULD still be used as streaming server and transcoding via videoengine would bring great performance increase.

We may then proceed with overlays for dmc params but still keep a single image.

We still have to see how stable the 4GB model is with mainline.
We may be forced to support legacy anyway if R4S 4GB is as stable in mainline as M4V2 is ;p
The good thing is that FriendlyElec provides (and hopefully supports) 5.4.y for R4S.

Reply:
`Rockchip ddr loader supports both types of memory in a single image and that's what we are using for most of the rk3399 family.`

So could you please tell me how to compile such a single image?